foohid/foohid_userclient.cpp

```diff
@@ -100,94 +143,127 @@ IOReturn it_unbit_foohid_userclient::sMethodList(it_unbit_foohid_userclient *tar
     return target->methodList(arguments);
 }
 
+IOReturn it_unbit_foohid_userclient::methodOpen() {
+    if (m_hid_provider == NULL || isInactive()) {
+        LogD("methodOpen->kIOReturnNotAttached");
+        return kIOReturnNotAttached;
+    }
+
+    if (!m_hid_provider->open(this)) {
```

Comment:
We probably don't want to call `open()` on the provider. The default IOService implementation only allows open open client at a time. I don't think we actually need `methodOpen` and `methodClose`. Instead, we should just do some of the cleanup from `methodClose` in `clientClose` or `didTerminate`. I can make those changes to this PR tomorrow.